### PR TITLE
Add feature to disable OMPT during runtime

### PR DESCRIPTION
**Pull request by cmcantalupo**
Add feature to OMPT disable during runtime

### CODE REVIEW

File: src/OMPT.cpp [hunk 1]
```diff
@@ -110,68 +136,32 @@ namespace geopm
...
-        if (g_curr_region_id != GEOPM_REGION_HASH_UNMARKED) {
-            geopm_prof_enter(g_curr_region_id);
-        }
-    }
+        uint64_t rid = region_id(parallel_function);
+        if (rid != GEOPM_REGION_HASH_UNMARKED) {
+            try {
+                m_prof.enter(rid);
+            }
+            catch (Exception) {
```
Comment (bakerbrandond) [inline src/OMPT.cpp]: Exception reference var missing.

File: src/OMPT.cpp [hunk 2]
```diff
@@ -110,68 +136,32 @@ namespace geopm
...
-    void ompt_finalize(ompt_data_t *data)
-    {
-
-    }
+        uint64_t rid = region_id(parallel_function);
+        if (rid != GEOPM_REGION_HASH_UNMARKED) {
+            try {
+                m_prof.exit(rid);
+            }
+            catch (Exception) {
```
Comment (bakerbrandond) [inline src/OMPT.cpp]: Exception reference again.

File: src/ompt_callback.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "OMPT.hpp"
+#include <omp-tools.h>
+
+
```
Comment (bakerbrandond) [inline src/ompt_callback.cpp]: config.h

File: src/Profile.hpp [hunk 4]
```diff
@@ -317,6 +318,8 @@ namespace geopm
             double m_overhead_time_startup;
             double m_overhead_time_shutdown;
     };
+
+    Profile &geopm_default_profile(void);
```
Comment (bakerbrandond) [inline src/Profile.hpp]: rm?

File: configure.ac [hunk 5]
```diff
@@ -460,23 +460,23 @@ fi
...
-  AC_CHECK_HEADER([gelf.h], [], [
-      echo "missing gelf.h: ELF library is required, use --with-libelf or --with-libelf-include to specify location"
-      exit -1])
 fi
 
 AC_SUBST([enable_ompt])
 AM_CONDITIONAL([ENABLE_OMPT], [test "x$enable_ompt" = "x1"])
 AM_CONDITIONAL([HAS_OMPT], [test "x$has_ompt" = "x1"])
 
+AC_SEARCH_LIBS([elf_begin], [elf])
```
[the remark this answers is not in the document]
Comment (cmcantalupo) [reply configure.ac]: done.  Fixed base and theta spec file.

File: src/OMPT.cpp (hunk 1 above)
Comment (cmcantalupo) [reply src/OMPT.cpp]: fixed

File: src/OMPT.cpp (hunk 2 above)
Comment (cmcantalupo) [reply src/OMPT.cpp]: fixed

File: src/ompt_callback.cpp (hunk 3 above)
Comment (cmcantalupo) [reply src/ompt_callback.cpp]: fixed

File: src/Profile.hpp (hunk 4 above)
Comment (cmcantalupo) [reply src/Profile.hpp]: fixed

**Pull request merged**